### PR TITLE
Sim 2552/htm

### PR DESCRIPTION
**Pull request by danielsf**
In order to facilitate efficient generation of simulated alerts, we needed tools at the clients-side to manipulate the Hierarchical Triangular Mesh scheme which we use as our spatial index on fatboy.  This pull request provides those tools.

(There may also be some small bugfixes that I noticed were needed in other parts of the code while working on this).

### CODE REVIEW

**Comment by rhiannonlynne** [comment]
This is pretty impressive. 

Mostly, readability for future maintenance and flake8 comments.

File: python/lsst/sims/utils/SpatialBounds.py [hunk 1]
```diff
@@ -167,17 +167,17 @@ def to_SQL(self, RAname, DECname):
 
         if np.abs(cosDec) > 1.0e-20:
             RAmax = self.RAdeg + \
-                360.0 * np.arcsin(np.sin(0.5 * self.radius) /
+                360.0 * np.arcsin(np.sin(self.radius) /
```
[the remark this answers is not in the document]
Comment (danielsf) [reply python/lsst/sims/utils/SpatialBounds.py]: No reason.  I probably wasn't as familiar with numpy when I first wrote this code.

File: python/lsst/sims/utils/htmModule.py [hunk 2]
```diff
@@ -0,0 +1,919 @@
...
+            return False
+        if self._level != other._level:
+            return False
+
+        tol = 1.0e-20
+
+        if not np.allclose(self._corners, other._corners, atol=tol):
+            return False
+
+        return True
```
[the remark this answers is not in the document]
Comment (danielsf) [reply python/lsst/sims/utils/htmModule.py]: Good point.

File: python/lsst/sims/utils/htmModule.py [hunk 3]
```diff
@@ -0,0 +1,919 @@
...
+        Returns
+        -------
+        A tuple:
+            Zeroth element is the unit vector pointing at
+            the center of the bounding circle
+
+            First element is the distance from the center of
+            the unit sphere to the plane of the bounding circle
+            (i.e. the dot product of the first element with the
+            most distant corner of the trixel).
```
[the remark this answers is not in the document]
Comment (danielsf) [reply python/lsst/sims/utils/htmModule.py]: Sorry.  It should read
```
First element is the distance from the center of
the unit sphere to the plane of the bounding circle
(i.e. the dot product of the zeroth element with the
most distant corner of the trixel).
```
Is that clearer?

**Pull request merged**